packages/appsync-modelgen-plugin/src/utils/process-auth.ts

```diff
@@ -51,6 +51,12 @@ export type AuthDirective = CodeGenDirective & {
   };
 };
 
+export const createIdentityClaim = (identityClaim: string): string => {
+  if (identityClaim === 'username' || identityClaim === 'sub::username') identityClaim = 'cognito:username';
```

Comment:
Codegen provides schema info for the libraries to use. There's both codegen specific logic that maps the schema's identityClaim to the codegenerated schema and then there's library logic that takes "cognito:username" and retrieves the "username" from the JWT access token's username. Since the new and future default identityClaim is "sub::username" and it produces backwards compatible APIs (subscription operations have `owner` input with the value being JWT access token's username value) then "cognito:username" as the output is the correct value for the libraries to continue to mapping "cognito:username" to the JWT accessToken.username.

Reply:
https://github.com/aws-amplify/amplify-android/blob/main/core/src/main/java/com/amplifyframework/core/model/AuthRule.java#L125-L134

>  An older version of the CLI incorrectly generated a value of "cognito:username" so we also check for this value and convert it to the proper default of "username" for backwards compatibility.

Reply:
So I think i'm coming to the conclusion that if we are introducing a new identityClaim who's value from the JWT token comes from "username" then "sub::username" (public facing) should return "username" (codegen library). We might as well fix the problem for username as well, by returning
```
// Ideal
  if identityClaim === 'sub::username' 
     identityClaim = 'username';
  // if "identityClaim == 'username' // do nothing, written here to show that it's a 1:1 mapping
     identityClaim == 'username'
``